src/js/components/Pagination/Pagination.js

```diff
@@ -199,29 +201,75 @@ const Pagination = forwardRef(
       ...navProps[control],
     }));
 
-    return (
-      <StyledPaginationContainer
-        flex={false}
-        {...theme.pagination.container}
-        {...rest}
-      >
-        <Nav
-          a11yTitle={ariaLabel || a11yTitle || 'Pagination Navigation'}
-          ref={ref}
+    let content;
```

Comment:
To avoid too much repeated code, can we try making what is currently being return assigned to content to start?

```
content = <StyledPaginationContainer
            flex={false}
            wrap
            {...theme.pagination.container}
            {...rest}
          >
            <Nav
              a11yTitle={ariaLabel || a11yTitle || 'Pagination Navigation'}
              ref={ref}
            >
              <Box as="ul" {...theme.pagination.controls}>
                {controls.map((control, index) => (
                  /* Using index as key (as opposed to a unique id) seems to
                   * help React prioritize rendering the updated controls as
                   * desired. Whereas, using a unique id resulted in rendering
                   * the active control with an undesired lag. */
                  // eslint-disable-next-line react/no-array-index-key
                  <PageControl key={index} size={size} {...control} />
                ))}
              </Box>
            </Nav>
          </StyledPaginationContainer>
```

Then we can do:

```
if (stepSelector) {
   content = <Box ...>
      <PaginationStep ... />
      {content}
      </Box>
}

Reply:
yes that makes sense Ill do the same with summary

Reply:
done